cli/config/configfile/file.go

```diff
@@ -300,7 +300,8 @@ func (configFile *ConfigFile) GetAllCredentials() (map[string]types.AuthConfig,
 	for registryHostname := range configFile.CredentialHelpers {
 		newAuth, err := configFile.GetAuthConfig(registryHostname)
 		if err != nil {
-			return nil, err
+			logrus.WithError(err).Warnf("Failed to get credentials for registry: %s", registryHostname)
```

Comment:
Was looking at this again, and while we should look at using the logger more, we should probably look at the output for it. Currently, the output looks like; 

```
WARN[0000] Failed to get credentials for registry: foo   error="something went wrong"
```

Still readable, but perhaps not as "friendly". I see elsewhere we just print our own, e.g.; https://github.com/docker/cli/blob/3ae101f41ef4b1ce7ed774bd0848780dffd07472/cli/config/config.go#L144

Which may look like;

```
WARNING: Error loading config file: /root/.docker/config.json: json: cannot unmarshal string into Go struct field ConfigFile.auths of type types.AuthConfig
```

Only issue is that this location of the code currently doesn't have a reference to the CLI's stdout or stderr, so if we change it, we probably need to hardcode it to `os.Stderr` (not ideal either).

Reply:
Ideas / suggestions (perhaps someone else has thoughts) welcome.

Don't think it's necessarily a blocker, if there's no better solution (for now)

Reply:
That's a good point! In general I think having the more standardized logging is good (vs just calling `fmt.Fprintf`), but the friendliness is a good point. Maybe we can look into seeing if we can configuring the output for it. I don't have better ideas right now though 😅

Reply:
Yes, there's a lot to look into in this area;

- There's `fmt.PrintXX` in many places
- While it's "generally" ok to use the defaults (I suspect `logrus` is not configured anywhere to use something other than `os.StdOut` / `os.StdErr`), ideally we'd have better controls for that
- ^^ sometimes we need to use a different approach depending on (e.g.) if an actual TTY is attached (see https://github.com/docker/cli/pull/3973)
- ^^ or (e.g.) if we want the output to be "quiet" (could be in the CLI itself, or other projects using parts of the CLI as a "library")
- I also went looking for _some_ of that, e.g. https://github.com/docker/cli/pull/3797
- (as mentioned) logrus (or other loggers) are great for logs, but the default output format often isn't great for printing "one-off", user-facing messages

So yes, this needs work (perhaps I should create a tracking ticket somewhere); e.g.

- Consider using a context-logger to pass a logger around
- That logger could (default) be a logger with a format that's "friendly enough"
- Look into best approach to pass around the streams (STDIN/STDOUT/STDERR), but when doing so, try to prevent coupling things too tightly (i.e., we may not want to pass cobra.Command _everywhere_).